sdks/aperture-py/aperture_sdk/flow.py

```diff
@@ -116,34 +119,42 @@ def set_result_cache(self, value: str, ttl: datetime.timedelta):
             "ttl": ttl,
         }
 
-        res: CacheUpsertResponse = self._fcs_stub.CacheUpsert(cache_upsert_request)
+        res = CacheUpsertResponse()
+        try:
+            res = self._fcs_stub.CacheUpsert(cache_upsert_request)
+        except grpc.RpcError as e:
+            self.logger.debug(f"Aperture gRPC call failed: {e.details()}")
 
-        if res.result_cache_response == None:
+        if res.result_cache_response is None:
             return KeyUpsertResponse(ValueError("No cache upsert response"))
 
         return KeyUpsertResponse(
             convert_cache_error(res.result_cache_response.error),
         )
 
-    async def delete_result_cache(self):
+    def delete_result_cache(self) -> KeyDeleteResponse:
         if not self._cache_key:
-            raise ValueError("No cache key")
+            return KeyDeleteResponse(ValueError("No cache key"))
 
         cache_delete_request = {
             "controlPoint": self._control_point,
             "key": self._cache_key,
         }
 
-        res: CacheDeleteResponse = self._fcs_stub.CacheDelete(cache_delete_request)
+        res = CacheDeleteResponse()
+        try:
+            res: CacheDeleteResponse = self._fcs_stub.CacheDelete(cache_delete_request)
+        except grpc.RpcError as e:
+            self.logger.debug(f"Aperture gRPC call failed: {e.details()}")
```

Comment:
Need to return KeyDeleteResponse here with appropriate error

Reply:
I don't see similar handling for upsert and lookup in this PR. You need to catch exception and return in those scenarios as well